statsd/statsd.go

```diff
@@ -592,16 +593,37 @@ func (c *Client) sendBlocking(m metric) error {
 	return worker.processMetric(m)
 }
 
-func (c *Client) sendToAggregator(mType metricType, name string, value float64, tags []string, rate float64, f bufferedMetricSampleFunc) error {
-	if c.aggregatorMode == channelMode {
+func (c *Client) sendToAggregator(mType metricType, name string, value float64, tags []string, rate float64, f bufferedMetricSampleFunc, bf bufferedMetricSampleBulkFunc, pool *sync.Pool) {
+	switch c.aggregatorMode {
+	case lossyMode:
+		if !shouldSample(rate) {
+			return
+		}
+
+		m := pool.Get().(*lossyBuffer)
```

Comment:
Isn't this creating a good risk of OOMs? The latency is basically gained here, where nothing would really block or make the client wait on every metric submission call, but isn't it at the cost of eventually creating a lot of lossy buffers here? Had you the chance to monitor/compare your app RAM usage while submitting the metrics? (there are different usage scenarios, yours may not be an issue here)